src/main/java/nonapi/io/github/classgraph/classpath/ClasspathFinder.java

```diff
@@ -121,7 +121,10 @@ public ClasspathFinder(final ScanSpec scanSpec, final LogNode log) {
         // If so, need to enable module scanning. If not, disable module scanning, since only the provided
         // classloader(s) should be scanned. (#382)
         boolean scanModules;
-        if (scanSpec.overrideClasspath != null) {
+        if (scanSpec.enableSystemJarsAndModules) {
+            // Always scan modules if system modules are enabled
+            scanModules = true;
+        } else if (scanSpec.overrideClasspath != null) {
```

Comment:
Won't this mean that if `scanSpec.enableSystemJarsAndModules` is true, all modules will be scanned, not just system modules? I'm rusty on my recollection of how this all works, but I think there would need to be some additional logic here that checks if any modules have been accepted for scanning, and if not, scanning should be limited to just system modules.

Reply:
I think that's accurate: the `ModuleFinder` was indiscriminately finding all modules, whether system modules or not. 718bcba addresses this such that `ModuleFinder` only records system modules if `scanSpec.enableSystemJarsAndModules==true` and only records non-system modules if a new `scanNonSystemModules` parameter is true.

By the same logic, it only sets `forceScanJavaClassPath` if `scanNonSystemModules` parameter is true, based on my limited understanding that all system modules are named. Consequently the commit also reverts the changes around `java.class.path` scanning that are no longer necessary.

To be fair, I don't have an environment with non-system or unnamed modules so I haven't been able to test this logic outside of the test suite.